### ISSUE TITLE
Fields display order in form view not the same as CRD

### ISSUE THREAD

**Issue by wuchaof** [openshift/console#8301]
We defined our fields in crd according to our published cr templated. Customer already familiar with the fields order. So we expect form view also display fields as the same order with we defined in CRD.
But actually form view will display the fields which been depended by other fields in the end of sibling fields.
like in crd we have
```
appversion: 21.0.1
ibm_license: true
new_feature_2101:    // this displayed when `appversion` is 21.0.1
```
in form view, it will show as 
```
ibm_license: true
appversion: 21.0.1
new_feature_2101: 
```

**Comment by wuchaof** [comment]
cc @lpchitta

**Comment by TheRealJon** [comment]
@wuchaof In what version are you seeing this issue? Are you using spec descriptors in your CSV as well? Is your CRD accessible somewhere that I can take a look so I can try to reproduce?

**Comment by TheRealJon** [comment]
cc @tlwu2013

**Comment by wuchaof** [comment]
@TheRealJon  I am using 4.6 OCP. We have spec descriptors in CSV like below.

```
specDescriptors:
  - path: new_feature_2101
     x-descriptors:
       - 'urn:alm:descriptor:com.tectonic.ui:fieldDependency:appversion: 21.0.1'
```

**Comment by TheRealJon** [comment]
@wuchaof We made some updates to the way fields are ordered in 4.6. See [this documentation](https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/README.md#ordering-of-form-fields) about how fields are ordered. 

TLDR:
Fields are sorted with the following priorities:
1. "required" fields with specDescriptors other than advanced
2. "required" fields without specDescriptors
3. "optional" fields with specDescriptors other than advanced
4. "optional" fields without specDescriptors
5. Fields with advanced specDescriptors wrapped in the 'Advanced Configuration' group

Judging from your issue description, this appears to be working as expected. In addition to the above rules, dependent fields are ordered directly after their control fields so that they remain close together on the form and the relationship can be seen. This might be a documentation issue since we didn't include the fieldDependency behavior in the above docs.

**Comment by TheRealJon** [comment]
Closing. Not a bug. @wuchaof Feel free to reopen with further comments if you feel this is still an issue.
